airflow/jobs/scheduler_job.py

```diff
@@ -636,8 +636,24 @@ def _process_executor_events(self, session: Session) -> int:
                 "run_start_date=%s, run_end_date=%s, "
                 "run_duration=%s, state=%s, executor_state=%s, try_number=%s, max_tries=%s, job_id=%s, "
                 "pool=%s, queue=%s, priority_weight=%d, operator=%s, queued_dttm=%s, "
-                "queued_by_job_id=%s, pid=%s"
+                "queued_by_job_id=%s, pid=%s, op_classpath=%s"
             )
+
+            # Adding op_class path for the task instance log in order to track where the
+            # operator/sensor originated from
+            # Get task from the Serialized DAG
+            try:
+                dag = self.dagbag.get_dag(ti.dag_id)
```

Comment:
This is, comparatively a _very_ expensive operation so I am wary of adding this feature like this without knowing how much it would impact performance in many large dags+high scheduling throughput cases.

Reply:
Actually this part of code was there already I made use of it for the log

Reply:
yes, but the original placement of the code would only be executed in a corner case, rather than every time.